controlplane/kubeadm/main.go

```diff
@@ -157,9 +158,20 @@ func setupReconcilers(ctx context.Context, mgr ctrl.Manager) {
 	if webhookPort != 0 {
 		return
 	}
+	// Set up a ClusterCacheTracker to provide to controllers
+	// requiring a connection to a remote cluster
+	tracker, err := remote.NewClusterCacheTracker(
+		ctrl.Log.WithName("remote").WithName("ClusterCacheTracker"),
+		mgr,
+	)
+	if err != nil {
+		setupLog.Error(err, "unable to create cluster cache tracker")
+		os.Exit(1)
```

Comment:
@furkatgofurov7 Given that this is a different main.go and separate binary, we should also setup the ClusterCacheReconciler to make this work properly see https://github.com/kubernetes-sigs/cluster-api/blob/master/main.go#L211-L228

Reply:
@vincepri thanks for suggestions, added. 

> We should make sure that's sufficient, and possibly expose options to uncache other certain objects. For example, we'd probably want to add appsv1.Deployment{} to the list of uncached objects, but probably only for KCP

I have a question, and if you can elaborate on this, it would be great. How can we make sure that it is sufficient, is the appsv1.Deployment{} only the one we want to add to uncached objects or there are more objects?

Reply:
We should look everywhere in the KCP codebase we use this new client and see which objects we're getting or listing

Reply:
@vincepri I have gone through the KCP codebase and found the following objects which make use of new client:

- [controlplanev1.KubeadmControlPlane{}](https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/controllers/controller.go#L107)
- [bootstrapv1.KubeadmConfig{}](https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/controllers/controller.go#L594)
- [corev1.Pod{}](https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/internal/workload_cluster_conditions.go#L364)
- [appsv1.Deployment{}](https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/internal/workload_cluster_coredns.go#L135)
- [appsv1.DaemonSet{}](https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/internal/workload_cluster.go#L387)

and plus ConfigMap && Secret which are currently the Objects being Uncached.

Reply:
Above listed are only the ones objects we are **getting** with client, following is list of  objects we are **listing** using client:

- [corev1.SecretList{}](https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/controllers/controller.go#L623)
- [clusterv1.MachineList{}](https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/internal/cluster.go#L75)
- [corev1.NodeList{}](https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/internal/workload_cluster.go#L99)

Reply:
```
corev1.Pod{}
appsv1.Deployment{}
appsv1.DaemonSet{}
```
These will need to be added, but we can probably follow-up in a new issue on that, because we'll need to expose this option in the tracker itself, instead of hardcoding them

Reply:
Thanks for clarification, sure thing, I can open a new issue and take care of it as well, if no one is interested.

Reply:
@vincepri FYI opened a follow-up issue #4146